torch/testing/_internal/common_optimizers.py

```diff
@@ -1011,6 +1012,14 @@ def _get_optim_inputs_including_global_cliquey_kwargs(
         optim_inputs_func=optim_inputs_func_adadelta,
         optim_error_inputs_func=optim_error_inputs_func_adadelta,
         supported_impls=("foreach", "differentiable"),
+        decorators=(
```

Comment:
You definitely don't need all these!

Reply:
I think I do, I only want to require cuda on the tests that use cuda ..

Reply:
You should be able to apply a skip on the instantiated cuda class def itself

Reply:
Yeah I'm going to submit a PR to use skipCUDAIf, (Jane and I discussed offline)